comp/core/tagger/taggerimpl/generic_store/store_test.go

```diff
@@ -11,206 +11,125 @@ import (
+	store := NewObjectStore[any]()
 
-	test := func(t *testing.T, isComposite bool) {
-		// initialise store
-		cfg := configmock.New(t)
-		cfg.SetWithoutSource("tagger.tagstore_use_composite_entity_id", isComposite)
-		store := NewObjectStore[any](cfg)
-
-		// store should be empty
-		assert.Equalf(t, store.Size(), 0, "store should be empty")
-
-		// add item to store
-		id := types.NewEntityID("prefix", "id")
-		store.Set(id, struct{}{})
-
-		// store size should be 1
-		assert.Equalf(t, 1, store.Size(), "store should contain 1 item")
+	// store should be empty
+	assert.Equalf(t, store.Size(), 0, "store should be empty")
 
-		// unset item
-		store.Unset(id)
+	// add item to store
+	id := types.NewEntityID("prefix", "id")
+	store.Set(id, struct{}{})
 
-		// store should be empty
-		assert.Equalf(t, 0, store.Size(), "store should be empty")
-	}
+	// store size should be 1
+	assert.Equalf(t, 1, store.Size(), "store should contain 1 item")
 
-	// default store
-	test(t, false)
+	// unset item
+	store.Unset(id)
 
-	// composite store
-	test(t, true)
+	// store should be empty
+	assert.Equalf(t, 0, store.Size(), "store should be empty")
 }
 
 func TestObjectStore_ListObjects(t *testing.T) {
-	test := func(t *testing.T, isComposite bool) {
-		// initialise store
-		cfg := configmock.New(t)
-		cfg.SetWithoutSource("tagger.tagstore_use_composite_entity_id", isComposite)
-		store := NewObjectStore[any](cfg)
-
-		// build some filter
-		fb := types.NewFilterBuilder()
-		fb.Include(types.EntityIDPrefix("prefix1"), types.EntityIDPrefix("prefix2"))
-		filter := fb.Build(types.HighCardinality)
-
-		// list should return empty
-		list := store.ListObjects(filter)
-		assert.Equalf(t, len(list), 0, "ListObjects should return an empty list")
-
-		// add some items
-		ids := []string{"prefix1://id1", "prefix2://id2", "prefix3://id3", "prefix4://id4"}
-		for _, id := range ids {
-			entityID, _ := types.NewEntityIDFromString(id)
-			store.Set(entityID, id)
-		}
-
-		// list should return empty
-		list = store.ListObjects(filter)
-		expectedListing := []any{"prefix1://id1", "prefix2://id2"}
-		assert.ElementsMatch(t, expectedListing, list)
+	store := NewObjectStore[any]()
+
+	// build some filter
+	fb := types.NewFilterBuilder()
+	fb.Include(types.EntityIDPrefix("prefix1"), types.EntityIDPrefix("prefix2"))
+	filter := fb.Build(types.HighCardinality)
+
+	// list should return empty
+	list := store.ListObjects(filter)
+	assert.Equalf(t, len(list), 0, "ListObjects should return an empty list")
+
+	// add some items
+	ids := []string{"prefix1://id1", "prefix2://id2", "prefix3://id3", "prefix4://id4"}
+	for _, id := range ids {
+		entityID, _ := types.NewEntityIDFromString(id)
+		store.Set(entityID, id)
 	}
 
-	// default store
-	test(t, false)
-
-	// composite store
-	test(t, true)
+	// list should return empty
+	list = store.ListObjects(filter)
+	expectedListing := []any{"prefix1://id1", "prefix2://id2"}
+	assert.ElementsMatch(t, expectedListing, list)
```

Comment:
Because we are running the callback function only on the items that match the filter that we created [here](https://github.com/DataDog/datadog-agent/blob/eabaac24351503b13e1e4dc0406aaaaa457f124c/comp/core/tagger/taggerimpl/generic_store/store_test.go#L92-L95):

```
	// build some filter
	fb := types.NewFilterBuilder()
	fb.Include(types.EntityIDPrefix("prefix1"), types.EntityIDPrefix("prefix2"))
	filter := fb.Build(types.HighCardinality)
```